File: 0485-max-consecutive-ones/0485-max-consecutive-ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        max_count = 0
        current_count = 0
        for i in range(0, len(nums)):
            if nums[i] == 1:
                current_count += 1
            if nums[i] == 0:
                max_count = current_count
                current_count = 0
        if max_count > current_count:
            return max_count
        else:
            return current_count

File: 0485-max-consecutive-ones/0485-max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int max_count = 0;
        int current_count = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] == 1) {
                current_count++;
            }
            if (nums[i] == 0) {
                max_count = max(max_count, current_count);
                current_count = 0;
            }
        }
        return max(max_count, current_count);
    }
};